nncf/experimental/post_training/algorithms/algorithm.py

```diff
@@ -52,6 +53,13 @@ class Algorithm(ABC):
 
     def __init__(self) -> None:
         self._model_transformer = None
+    
+    @property
+    @abstractmethod
+    def available_backends(self) -> List[BackendType]:
+        """
+        Returns list of the avaliable backends for the algorithm
+        """
 
     @property
     def model_transformer(self) -> StaticModelTransformerBase:
```

Comment:
This was made just to keep the `apply` method signature as is and to have a model transformer in each method of the algorithm. But, since the algorithm signature was updated, we can update the `apply` method with the new variable, if needed.

Reply:
Do we have a chance to simplify logic here? Can we create ```ModelTransformer``` in each instance of algorithm?

Reply:
Yes, we can create the new `ModelTransformer` in each algorithm. I will remove this unnecessary optimization.